src/background/services/tabEvents.ts

```diff
@@ -1,16 +1,55 @@
+import { type Browser, runtime } from 'webextension-polyfill'
 import { MonetizationService } from './monetization'
+import { StorageService } from './storage'
+import { IsTabMonetizedPayload } from '@/shared/messages'
 
-export class TabEvents {
-  constructor(private monetizationService: MonetizationService) {}
-
-  onRemovedTab = (tabId: number) => {
-    this.monetizationService.clearTabSessions(tabId)
-  }
+const iconActive34 = runtime.getURL('assets/icons/icon-active-34.png')
+const iconActive128 = runtime.getURL('assets/icons/icon-active-128.png')
+const iconInactive34 = runtime.getURL('assets/icons/icon-inactive-34.png')
+const iconInactive128 = runtime.getURL('assets/icons/icon-inactive-128.png')
+const iconWarning34 = runtime.getURL('assets/icons/icon-warning-34.png')
+const iconWarning128 = runtime.getURL('assets/icons/icon-warning-128.png')
 
+export class TabEvents {
+  constructor(
+    private monetizationService: MonetizationService,
+    private storage: StorageService,
+    private browser: Browser
+  ) {}
   // TODO: This is not ideal. Find a better way to clear the sessions for a specific tab.
   // When closing the tab, we receive the STOP_MONETIZATION message as well.
   // Maybe check if the tab is closed in the content script?
-  onUpdatedTab = (tabId: number) => {
+  onRemovedTab = (tabId: number) => {
```

Comment:
Make sure to merge the `main` branch first. Both `onUpdatedTab` and `onRemovedTab` are being used right now.